restore-hooks_product-requirements.md

```diff
@@ -0,0 +1,337 @@
+Velero Restore Hooks - PRD (Product Requirements Document) 
+
+MVP Feature Set
+
+Relates to:  [https://github.com/vmware-tanzu/velero/issues/2116](https://github.com/vmware-tanzu/velero/issues/2116)
+
+
+# **Change tracking**
+
+_This is a live document, you can reach me on the following channels for more information or for any questions:_
+
+
+
+*   _Email:  [bstephanie@vmware.com](mailto:bstephanie@vmware.com)_
+*   _Kubernetes slack:  _
+    *   _Channel:  #velero_
+    *   _@Stephanie Bauman_
+*   _VMware internal slack:_
+    *   _@bstephanie_
+
+
+# **Relates to Git Issues:**
+
+
+
+*   [https://app.zenhub.com/workspaces/velero-5c59c15e39d47b774b5864e3/issues/vmware-tanzu/velero/2465](https://app.zenhub.com/workspaces/velero-5c59c15e39d47b774b5864e3/issues/vmware-tanzu/velero/2465)
+    *   [https://github.com/vmware-tanzu/velero/pull/2465/files?short_path=140e0c6#diff-140e0c6b370f250ee97f6ecafc8dbb7a](https://github.com/vmware-tanzu/velero/pull/2465/files?short_path=140e0c6#diff-140e0c6b370f250ee97f6ecafc8dbb7a)
+*   [https://github.com/vmware-tanzu/velero/issues/1150](https://github.com/vmware-tanzu/velero/issues/1150)
+
+Background
+
+Velero supports restore operations but there are gaps in the process.  Gaps in the restore process require users to manually carry out steps to start, clean up, and end the restore process. Other gaps in the restore process can cause issues with application performance for applications running in a pod when a restore operation is carried out. 
+
+On a restore, Velero currently does not include hooks to execute a pre- or -post restore script. As a result, users are required to perform additional actions following a velero restore operation. Some gaps that currently exist in the Velero restore process are:
+
+
+
+*   Users can create a restore operation but has no option to customize or automate commands during the start of the restore operation
+*   Users can perform post restore operations but have no option to customize or automate commands during the end of the restore operation
+
+Strategic Fit
+
+Adding a restore hook action today would allow Velero to unpack the data that was backed up in an automated way by enabling Velero to execute commands in containers during a restore. This will also improve the restore operations on a container and mitigate against any negative performance impacts of apps running in the container during restore.
+
+Purpose / Goal
+
+The purpose of this feature is to improve the extensibility and user experience of pre and post restore operations for Velero users.
```

Comment:
Is there a notion of pre and post for restore hooks? Before what would we run a pre hook and after what would post hook?

Reply:
The use cases detail the requirements. This level of detail won't be found in the strategic fit/goals section.

Reply:
Per discussion w/ Ashish - we can run restore hooks during a restore or after a restore. Running the velero restore command is not part of the notion of a hook.  

Will be calling this a post hook.